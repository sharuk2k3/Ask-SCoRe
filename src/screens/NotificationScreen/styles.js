import {StyleSheet, Dimensions} from "react-native";
import {
  COLOR_PRIMARY,
  COLOR_GRAY_SECONDARY,
  COLOR_GRAY_PRIMARY,
} from "../../config/styles";

let deviceWidth = Dimensions.get("window").width;
let deviceHeight = Dimensions.get("window").height;

export default StyleSheet.create({
  WholeTabBar: {
    width: "100%",
    height: 60,
    backgroundColor: COLOR_PRIMARY,
    alignSelf: "center",
    borderRadius: 3,
    flexDirection: "row",
    paddingVertical: 5,
  },

  LoginTabBar: {
    height: "100%",
    width: "50%",
    backgroundColor: COLOR_PRIMARY,
    borderRadius: 20,
  },

  SignupTabBar: {
    height: "100%",
    width: "50%",
    backgroundColor: COLOR_PRIMARY,
    borderRadius: 3,
  },

  BarStyle: {
    textAlign: "center",
    marginTop: 8,
    color: COLOR_PRIMARY,
    fontSize: 20,
    fontWeight: "800",
  },

  BarStyle1: {
    textAlign: "center",
    marginTop: 8,
    color: COLOR_PRIMARY,
    fontSize: 20,
    fontWeight: "800",
  },

  ViewStyle: {
    backgroundColor: "white",
    flex: 1,
    justifyContent: "flex-start",
  },

  DividerView: {
    borderBottomColor: "#23598B",
    borderBottomWidth: 7,
    marginVertical: 3,
    marginHorizontal: 0,
  },

  DividerView1: {
    borderBottomColor: COLOR_PRIMARY,
    borderBottomWidth: 2,
    marginVertical: 3,
    marginHorizontal: 0,
  },

  SafeAreaStyle: {flex: 0, backgroundColor: COLOR_PRIMARY},

  IconStyle: {color: COLOR_GRAY_PRIMARY, alignSelf: "center"},

  IconStyle1: {color: "white", alignSelf: "center"},

  TextStyle: {alignSelf: "center", color: "white"},
});
