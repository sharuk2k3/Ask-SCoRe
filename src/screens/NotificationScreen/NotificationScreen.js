import React, {Component} from "react";
import {View, Text, TouchableOpacity, SafeAreaView} from "react-native";
import MediumActivity from "../MediumActivity/MediumActivity";
import GithubActivity from "../GithubActivity/GithubActivity";
import Icon from "react-native-vector-icons/Ionicons";
import styles from "./styles";

class NotificationScreen extends Component {
  state = {showingup: true};
  renderContent() {
    switch (this.state.showingup) {
      case true:
        return (
          <View style={{flex: 1}}>
            <GithubActivity />
          </View>
        );
      case false:
        return (
          <View>
            <MediumActivity />
          </View>
        );
      default:
        return <Text>Error</Text>;
    }
  }

  render() {
    return (
      <>
        <SafeAreaView style={styles.SafeAreaStyle} />
        <View style={styles.ViewStyle}>
          <View style={styles.WholeTabBar}>
            {this.state.showingup ? (
              <TouchableOpacity
                style={styles.SignupTabBar}
                onPress={() => {
                  this.setState({showingup: true});
                }}>
                <View>
                  <Icon
                    style={styles.IconStyle1}
                    size={26}
                    name={"logo-github"}
                  />
                  <Text style={styles.TextStyle}>Github</Text>
                </View>
                <View style={styles.DividerView} />
              </TouchableOpacity>
            ) : (
              <TouchableOpacity
                style={styles.LoginTabBar}
                onPress={() => {
                  this.setState({showingup: true});
                }}>
                <View>
                  <Icon
                    style={styles.IconStyle}
                    size={26}
                    name={"logo-github"}
                  />
                  <Text style={styles.TextStyle}>Github</Text>
                </View>
                <View style={styles.DividerView1} />
              </TouchableOpacity>
            )}
            {this.state.showingup === false ? (
              <TouchableOpacity
                style={styles.SignupTabBar}
                onPress={() => {
                  this.setState({showingup: false});
                }}>
                <View>
                  <Icon
                    style={styles.IconStyle1}
                    size={26}
                    name={"newspaper-outline"}
                  />
                  <Text style={styles.TextStyle}>Medium</Text>
                </View>
                <View style={styles.DividerView} />
              </TouchableOpacity>
            ) : (
              <TouchableOpacity
                style={styles.LoginTabBar}
                onPress={() => {
                  this.setState({showingup: false});
                }}>
                <View>
                  <Icon
                    style={styles.IconStyle}
                    size={26}
                    name={"newspaper-outline"}
                  />
                  <Text style={styles.TextStyle}>Medium</Text>
                </View>
                <View style={styles.DividerView1} />
              </TouchableOpacity>
            )}
          </View>
          <View style={{flex: 1}}>{this.renderContent()}</View>
        </View>
      </>
    );
  }
}

export default NotificationScreen;
