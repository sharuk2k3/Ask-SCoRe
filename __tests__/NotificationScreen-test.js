import "react-native";
import React from "react";
import NotificationScreen from "../src/screens/NotificationScreen/NotificationScreen";
import renderer from "react-test-renderer";

jest.useFakeTimers();

test("renders Notification Screen correctly", async () => {
  const tree = renderer.create(<NotificationScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
