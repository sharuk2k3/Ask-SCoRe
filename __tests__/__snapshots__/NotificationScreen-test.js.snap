// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Notification Screen correctly 1`] = `
Array [
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#51AD28",
        "flex": 0,
      }
    }
  />,
  <View
    style={
      Object {
        "backgroundColor": "white",
        "flex": 1,
        "justifyContent": "flex-start",
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#51AD28",
          "borderRadius": 3,
          "flexDirection": "row",
          "height": 60,
          "paddingVertical": 5,
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#51AD28",
            "borderRadius": 3,
            "height": "100%",
            "opacity": 1,
            "width": "50%",
          }
        }
      >
        <View>
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 26,
                },
                Object {
                  "alignSelf": "center",
                  "color": "white",
                },
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "white",
              }
            }
          >
            Github
          </Text>
        </View>
        <View
          style={
            Object {
              "borderBottomColor": "#23598B",
              "borderBottomWidth": 7,
              "marginHorizontal": 0,
              "marginVertical": 3,
            }
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#51AD28",
            "borderRadius": 20,
            "height": "100%",
            "opacity": 1,
            "width": "50%",
          }
        }
      >
        <View>
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 26,
                },
                Object {
                  "alignSelf": "center",
                  "color": "#dbdbdb",
                },
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "white",
              }
            }
          >
            Medium
          </Text>
        </View>
        <View
          style={
            Object {
              "borderBottomColor": "#51AD28",
              "borderBottomWidth": 2,
              "marginHorizontal": 0,
              "marginVertical": 3,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RCTSafeAreaView
          emulateUnlessSupported={true}
          style={
            Object {
              "backgroundColor": "#e6e6e6",
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "marginBottom": 15,
                }
              }
            />
            <RCTScrollView
              ListFooterComponent={
                <View
                  style={
                    Object {
                      "height": 50,
                    }
                  }
                />
              }
              data={Array []}
              disableVirtualization={false}
              getItem={[Function]}
              getItemCount={[Function]}
              horizontal={false}
              initialNumToRender={10}
              keyExtractor={[Function]}
              maxToRenderPerBatch={10}
              onContentSizeChange={[Function]}
              onEndReachedThreshold={2}
              onLayout={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              removeClippedSubviews={false}
              renderItem={[Function]}
              scrollEventThrottle={50}
              stickyHeaderIndices={Array []}
              updateCellsBatchingPeriod={50}
              viewabilityConfigCallbackPairs={Array []}
              windowSize={21}
            >
              <View>
                <View
                  onLayout={[Function]}
                >
                  <View
                    style={
                      Object {
                        "height": 50,
                      }
                    }
                  />
                </View>
              </View>
            </RCTScrollView>
          </View>
        </RCTSafeAreaView>
      </View>
    </View>
  </View>,
]
`;
